src/components/Button/UniversalButton/UniversalButton.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import styled, { css } from "styled-components";
+import {
+    buttonSizes,
+    typographyStyles,
+    buttonVariants,
+    buttonColors,
+    ButtonProps
+} from "@/assets/styles/constants/buttonConstants";
+import { ButtonVariant, ButtonSize } from "@/assets/styles/constants/buttonConstants";
+
+const ButtonStyled = styled.button<{
+    $variant?: ButtonVariant;
+    $size?: ButtonSize;
+    $buttonColor?: string;
+    $dotColor?: string;
+}>`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 0 16px;
+    height: ${({ $size }) => buttonSizes[$size || "medium"]};
+    background-color: ${({ $variant, $buttonColor }) =>
+            $variant === buttonVariants.outline ||
+            $variant === buttonVariants.dotted ||
+            $variant === buttonVariants.glowing
+                    ? "transparent"
+                    : $buttonColor || buttonColors[$variant || "primary"]};
+    color: ${({ $variant, $buttonColor }) =>
+            $variant === buttonVariants.outline ||
+            $variant === buttonVariants.dotted ||
+            $variant === buttonVariants.glowing
+                    ? $buttonColor || buttonColors.primary
+                    : "#fff"};
+    border: ${({ $variant, $buttonColor }) =>
+            $variant === buttonVariants.outline ||
+            $variant === buttonVariants.dotted ||
+            $variant === buttonVariants.glowing
+                    ? `2px solid ${$buttonColor || buttonColors.primary}`
+                    : "none"};
+    border-radius: 8px;
+    font-weight: 500;
+    cursor: ${({ $variant }) => ($variant === buttonVariants.disabled ? "not-allowed" : "pointer")};
+    opacity: ${({ $variant }) => ($variant === buttonVariants.disabled ? 0.6 : 1)};
+    transition: all 0.3s ease;
+
+    &:hover {
+        opacity: ${({ $variant }) => ($variant === buttonVariants.disabled ? 0.6 : 0.8)};
+    }
+
+    ${({ $variant, $dotColor }) =>
+            $variant === buttonVariants.dotted &&
+            css`
+                gap: 16px;
+                &::before {
+                    content: "•";
+                    font-size: 20px;
+                    color: ${$dotColor || buttonColors.primary};
+                }
+            `}
+
+    ${({ $variant }) =>
+            $variant === buttonVariants.glowing &&
+            css`
+                outline: 3px solid ${buttonColors.glowingBorder};
+                outline-offset: 0px;
+            `}
+`;
```

Comment:
why styles are here? they took 68 lines of component, its a looot, move to styles, file